src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -0,0 +1,222 @@
+
+            db_results[key] = db_obj.id
+            reverse_db_results[db_obj.id] = db_obj.string
+
+        indexer_cache.set_many(db_results)
+        indexer_cache.set_many(reverse_db_results)
+        return db_results
+
+    def _get_many_ids(self, keys: Sequence[str]) -> MutableMapping[str, Optional[int]]:
+        """
+        Takes a list of keys formatted as such "org_string:string". Ex. "1:release"
+
+        Returns a mapping of the key to the id (or None if the id doesn't exist):
+            {
+                "1:release": 3,
+                "2:v1": None
+            }
+
+        This method first attempts to check memcache to see if we have the id
+        there and if not, we attempt to look up records in the db. We record
+        the cache hits and misses here, but the misses don't mean that the
+        records were in the db, just means they weren't in the cache.
+        """
+        final_results: MutableMapping[str, Optional[int]] = {}
+        cache_results = indexer_cache.get_many(keys)
+
+        db_look_up_keys = set()
+        for key in keys:
+            result = cache_results.get(key)
+            if isinstance(result, int):
+                final_results[key] = result
+            else:
+                final_results[key] = None
+                db_look_up_keys.add(key)
+
+        cache_hits = len(final_results.keys())
+        cache_misses = len(db_look_up_keys)
+
+        metrics.incr(
+            _INDEXER_CACHE_METRIC,
+            tags={"cache_hit": "true", "caller": "get_many_ids"},
+            amount=cache_hits,
+        )
+        metrics.incr(
+            _INDEXER_CACHE_METRIC,
+            tags={"cache_hit": "false", "caller": "get_many_ids"},
+            amount=cache_misses,
+        )
+
+        # if we found everything we need in the cache, return early
+        # otherwise attempt to look up records in db (and cache results
+        # if we find them in the db)
+        if len(db_look_up_keys) == 0:
+            return final_results
+
+        db_results = self._get_db_ids_and_set_cache(list(db_look_up_keys))
+        if db_results:
+            final_results.update(db_results)
+
+        return final_results
+
+    def bulk_record(
+        self, org_strings: MutableMapping[int, Set[str]]
+    ) -> MutableMapping[int, MutableMapping[str, int]]:
+        keys: List[str] = []
+        for org_id in org_strings:
+            keys.extend([f"{org_id}:{string}" for string in org_strings[org_id]])
+
+        results: MutableMapping[int, MutableMapping[str, int]] = defaultdict(dict)
+        keys_to_ids = self._get_many_ids(keys)
+
+        unmapped: List[str] = []
+        for key, id in keys_to_ids.items():
+            if id is None:
+                unmapped.append(key)
+                continue
+            org, string = key.split(":")
+            results[int(org)][string] = id
+
+        if not unmapped:
+            return results
+
+        new_mapped = self._record_many_strings(unmapped)
```

Comment:
COuld you please still record `_INDEXER_CACHE_FETCH_METRIC` to compare the old metrics with the new ones when you rollout.